### DISCUSSION TITLE
Intercept traffic in `local` mode, then forward to upstream proxy

### DISCUSSION THREAD

**Discussion by lawrence-forooghian** [mitmproxy/mitmproxy#6786] [Q&A] [answered]
I would like to implement the following behaviour:

1. Use mitmproxy’s `local` mode in order to transparently intercept all traffic from a given process and to handle TLS termination, then
2. Instruct this instance of mitmproxy to forward all intercepted traffic to another proxy server.

As an example, I naïvely thought that `mitmproxy --mode local:curl --mode upstream:http://localhost:8081` might allow me to capture all `curl` traffic and forward it to a proxy running at `http://localhost:8081`. However, when I run a proxy at `http://localhost:8081` using `mitmproxy --listen-port 8081`, this second instance of mitmproxy doesn’t display any traffic, suggesting that the first instance is not forwarding traffic.

Is there something I’m misunderstanding here?

**Comment by mhils** [comment, marked as answer]
> `mitmproxy --mode local:curl --mode upstream:http://localhost:8081`

This commands starts two proxy modes: First, proxy all local cURL traffic. Second, spawn an HTTP proxy on port 8080 that will forward all traffic to another HTTP proxy upstream at `localhost:8081`. Proxy modes are not "interconnected".
There currently is no (easy) way to use an upstream proxy with local redirect mode.

**Reply by lawrence-forooghian** [reply]
Got it, thank you!

> There currently is no (easy) way to use an upstream proxy with local redirect mode.

I’m assuming that means "even if you write an add-on"?

**Comment by vitaly-ps** [comment]
I've submitted a PR that seems to fix the issue being disucussed here: https://github.com/mitmproxy/mitmproxy/pull/7985
Hope this helps. Upvote if it is helpful.
